utils/dynamics.go

```diff
@@ -1,20 +1,61 @@
 package utils
 
-import "github.com/alicenet/alicenet/bridge/bindings"
+import (
+	"errors"
+	"strconv"
+	"strings"
 
-func CompareCanonicalVersion(newVersion bindings.CanonicalVersion) (bool, bool, bool, bindings.CanonicalVersion) {
-	localVersion := GetLocalVersion()
+	"github.com/alicenet/alicenet/bridge/bindings"
+	"github.com/alicenet/alicenet/config"
+)
+
+var (
+	ErrDevelopmentVersion  = errors.New("not tagged build, running in developer mode")
+	ErrUnexpectedTagFormat = errors.New("unexpected version format. expected vX.Y.Z[-optional]")
+)
+
+func CompareCanonicalVersion(newVersion bindings.CanonicalVersion) (bool, bool, bool, bindings.CanonicalVersion, error) {
+	localVersion, err := GetLocalVersion()
 	return newVersion.Major > localVersion.Major,
 		newVersion.Major == localVersion.Major && newVersion.Minor > localVersion.Minor,
 		newVersion.Major == localVersion.Major && newVersion.Minor == localVersion.Minor && newVersion.Patch > localVersion.Patch,
-		localVersion
+		localVersion, err
 }
 
-func GetLocalVersion() bindings.CanonicalVersion {
-	return bindings.CanonicalVersion{
-		Major:      0,
-		Minor:      0,
-		Patch:      0,
-		BinaryHash: [32]byte{},
+func GetLocalVersion() (bindings.CanonicalVersion, error) {
+	var canonicalVersion bindings.CanonicalVersion
+
+	version := config.Configuration.Version
+	if version == "dev" {
+		return canonicalVersion, ErrDevelopmentVersion
```

Comment:
This logic concerns me because it may cause a failure of the node to run or a failure of the monitoring task to work as expected. I want to discuss with you before merge.